src/NosCore.Controllers/GroupPacketController.cs

```diff
@@ -328,22 +327,27 @@ public void LeaveGroup(PleavePacket __)
 
             if (group.Count > 2)
             {
-                Session.Character.LeaveGroup();
-
                 if (group.IsGroupLeader(Session.Character.CharacterId))
                 {
+                    Session.Character.LeaveGroup();
+
                     var session = Broadcaster.Instance.GetCharacter(s =>
-                        s.VisualId == group.Values.First().Item2.VisualId);
+                        s.VisualId == group.Values.OrderBy(t => t.Item1).FirstOrDefault(u =>
+                        u.Item2.VisualType == VisualType.Player).Item2.VisualId);
 
                     if (session == null)
                     {
                         return;
                     }
 
-                    Broadcaster.Instance.Sessions.SendPacket(new InfoPacket
```

Comment:
everyone in the group is suppose to receive the message

Reply:
That's not how works official, but then we need to change the message to something like: "The party master is [{0}]".

Reply:
are you sure ?

Reply:
xD, I have coded this today basing it in the official US-Cylloan (using 3 acc at the same time and making sure all the packets are the same ;) )